### PR TITLE
toFormData helper function

### PR DESCRIPTION
**Pull request by carpben**
With this functionality we can provide the user an option to pass a data object, instead of building FormData by himself. When the data is nested, building form data isn't trivial, and can result in an inappropriate structure and bugs. Hence usage of this functionality seems more reliable. 

The function can handle nested data of the following types: object, array, string, number, boolean, null, undefined, Fiile. 

Related feature request thread: https://github.com/axios/axios/issues/3721

This doesn't deal with the API in which this functionality will be offered to the user.

### CODE REVIEW

**Review by steph931 — APPROVED** [review]

**Comment by carpben** [comment]
Commit was done by mistake with my previous company git user. Let me try again.

**Comment by carpben** [comment]
@steph931 I submitted a newer PR https://github.com/axios/axios/pull/3757 it is identical, but commits are done by my user.
